docs/architecture/adr-043-nft-module.md

```diff
@@ -35,341 +36,224 @@ The current design is based on the work done by [IRISnet team](https://github.co
 
 We will create a module `x/nft`, which contains the following functionality:
 
-- Store and transfer NFTs utilizing `x/bank`.
-- Mint and burn NFTs.
-- Use `x/authz` to implement ERC721-style authorization.
+- Store NFTs and track their ownership.
+- Expose `Keeper` interface for composing modules to mint and burn NFTs.
+- Expose external `Message` interface for users to transfer ownership of their NFTs.
 - Query NFTs and their supply information.
 
 ### Types
 
-#### Metadata
+#### Genre
 
-We define a `Metadata` model for `NFT Type`, which is comparable to an ERC721 contract on Ethereum, under which a collection of `NFT`s can be created and managed.
+We define a model for NFT **Genre**, which is comparable to an ERC721 Contract on Ethereum, under which a collection of NFTs can be created and managed.
 
 ```protobuf
-message Metadata {
-  string type          = 1; // Required, unique key, alphanumeric
-  string name          = 2;
-  string symbol        = 3;
-  string description   = 4;
-  bool mint_restricted = 5;
-  bool edit_restricted = 6;
+message Genre {
+  string id              = 1;
+  string name            = 2;
+  string symbol          = 3;
+  string description     = 4;
+  string uri             = 5;
+  bool mint_restricted   = 10;
+  bool update_restricted = 11;
 }
 ```
 
-- The `type` is the identifier of the NFT type/class.
-- The `name` is a descriptive name of this NFT type.
-- The `symbol` is the symbol usually shown on exchanges for this NFT type.
-- The `description` is a detailed description of this NFT type.
-- The `mint_restricted` flag, if set to true, indicates that only the issuer of this type can mint NFTs for it.
-- The `edit_restricted` flag, if set to true, indicates that NFTs of this type cannot be edited once minted.
+- `id` is an alphanumeric identifier of the NFT genre; it is used as the primary index for storing the genre; _required_
+- `name` is a descriptive name of the NFT genre; _optional_
+- `symbol` is the symbol usually shown on exchanges for the NFT genre; _optional_
+- `description` is a detailed description of the NFT genre; _optional_
+- `uri` is a URL pointing to an off-chain JSON file that contains metadata about this NFT genre ([OpenSea example](https://docs.opensea.io/docs/contract-level-metadata)); _optional_
+- `mint_restricted` flag, if set to true, indicates that only the genre owner can mint NFTs, otherwise anyone can do so; _required_
+- `udpate_restricted` flag, if set to true, indicates that no one can update NFTs, otherwise only NFT owners can do so; _required_
 
 #### NFT
 
 We define a general model for `NFT` as follows.
 
 ```protobuf
 message NFT {
-  string type              = 1; // The type of this NFT
-  string id                = 2; // The identifier of this NFT
-  string uri               = 3; 
-  google.protobuf.Any data = 4;
+  string genre = 1;
+  string id    = 2;
+  string uri   = 3;
+  string data  = 10;
```

Comment:
what was the reason to change from `protobuf.Any` to `string`?

Reply:
> what was the reason to change from `protobuf.Any` to `string`?

Typo, will change back to protobuf.Any